cognite/neat/explorer/explorer.py

```diff
@@ -357,41 +358,101 @@ def get_nodes_and_edges(request: NodesAndEdgesRequest):
-    nodes_query = f"SELECT DISTINCT ?node_id ?node_class ?node_name WHERE \
-    {{ {nodes_filter} \
-    ?node_id {node_name_property} ?node_name . ?node_id rdf:type ?node_class }} "
-    if len(request.src_edge_filter) > 0:
-        edges_src_filter = "VALUES ?src_object_class { " + " ".join([f"<{x}>" for x in request.src_edge_filter]) + " }"
-    if len(request.dst_edge_filter) > 0:
-        edges_dst_filter = "VALUES ?dst_object_class { " + " ".join([f"<{x}>" for x in request.dst_edge_filter]) + " }"
-
-    edges_query = f"SELECT  ?src_object_ref ?conn ?dst_object_ref \
-        WHERE {{ \
-        {edges_src_filter} \
-        {edges_dst_filter} \
-        ?src_object_ref ?conn  ?dst_object_ref . \
-        ?src_object_ref  rdf:type ?src_object_class . \
-        ?dst_object_ref  rdf:type ?dst_object_class .\
-        }} "
-    logging.info(f"Nodes query : {nodes_query}")
-    logging.info(f"Edges query : {edges_query}")
-    nodes_result = get_data_from_graph(nodes_query, request.graph_name, request.workflow_name)
-    edges_result = get_data_from_graph(edges_query, request.graph_name, request.workflow_name)
+    nodes_result = {}
+    edges_result = {}
+    query = ""
+    elapsed_time_sec = 0
+    """
+    "nodes": [
+        {
+            "node_id": "http://purl.org/nordic44#_a63ce14e-fba0-4f9e-8b59-7ef2fe887ff8",
+            "node_class": "http://entsoe.eu/CIM/SchemaExtension/3/2#RateTemperature",
+            "node_name": "-5degreeCelsius"
+        }],
+    "edges": [
+        {
+            "src_object_ref": "http://purl.org/nordic44#_f1769eaa-9aeb-11e5-91da-b8763fd99c5f",
+            "conn": "http://iec.ch/TC57/2013/CIM-schema-cim16#OperationalLimitSet.Terminal",
+            "dst_object_ref": "http://purl.org/nordic44#_2dd90186-bdfb-11e5-94fa-c8f73332c8f4"
+        },
+    """
+    if request.sparql_query:
+        mixed_result = get_data_from_graph(request.sparql_query, request.graph_name, request.workflow_name)
+
+        edges_result = []
+        nodes_result = []
+        try:
+            nodes_result = [
+                {
+                    "node_id": v[rdflib.Variable("node_id")],
+                    "node_class": v[rdflib.Variable("node_class")],
+                    "node_name": v[rdflib.Variable("node_name")],
+                }
+                for v in mixed_result["rows"]
+            ]
+        except Exception as e:
+            logging.error(f"Error while parsing nodes : {e}")
+
+        try:
+            edges_result = [
+                {
+                    "src_object_ref": v[rdflib.Variable("src_object_ref")],
+                    "dst_object_ref": v[rdflib.Variable("dst_object_ref")],
+                }
+                for v in mixed_result["rows"]
+                if rdflib.Variable("src_object_ref") in v and rdflib.Variable("dst_object_ref") in v
+            ]
+        except Exception as e:
+            logging.error(f"Error while parsing edges : {e}")
```

Comment:
Not sure if this makes sense, but should these be a helper functions? Something like `parse_node_results(mixed_results)` and `parse_edge_results(mixed_results)`.

Reply:
Good point